.github/workflows/test_run.yaml

```diff
@@ -60,5 +52,31 @@ jobs:
 
             - uses: codecov/codecov-action@v1
               with:
-                fail_ci_if_error: false
-                token: ${{ secrets.CODECOV_TOKEN }}
+                  fail_ci_if_error: false
+                  token: ${{ secrets.CODECOV_TOKEN }}
+
+    integration-tests:
```

Comment:
I wonder if that should be a separate github action with it's own triggers?
Let's say we run it only on each push to development and main branches?